03-components/50-weather-components/WeatherList.js

```diff
@@ -0,0 +1,42 @@
+import { defineComponent } from "vue";
+import WeatherListItem from "./WeatherListItem";
+
+export default defineComponent({
+  name: 'WeatherList',
+
+  components: {
+    WeatherListItem
+  },
+
+  props: {
+    weatherData: {
+      type: Array,
+      required: true
+    },
+
+    weatherConditionIcons: {
+      type: Object,
+      required: true
+    },
```

Comment:
To use all these components (even the list) a developer has to pass the icons object. This is inconvenient. A developer must have the object with the icons, pass it in any of these components.

This could be useful as an additional feature to pass custom icons. But only having the default value, that works out of the box.

Reply:
So it is better to import WeatherConditionIcons inside of WeatherList. Right?